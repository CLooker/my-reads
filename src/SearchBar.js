import React from 'react';
import { Link } from 'react-router-dom';
import { DebounceInput } from 'react-debounce-input';
import './App.css';

const SearchBar = function(props) {
    return (
      <div className="search-books">
        <div className="search-books-bar">
          <Link
            className="close-search"
            onClick={props.closeSearch}
            to="/"
          >
            Close
          </Link>
          <div className="search-books-input-wrapper">
              <DebounceInput
                minLength={1}
                debounceTimeout={300}
                type="text"
                value={props.query}
                placeholder="Search by title or author"
                onChange={props.handleChange}
              />
          </div>
        </div>
        <div className="search-books-results">
          <ol className="books-grid"></ol>
        </div>
      </div>
    )
}

export default SearchBar;