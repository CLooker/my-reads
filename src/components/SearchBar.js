import React from 'react';
import { Link } from 'react-router-dom';
import { DebounceInput } from 'react-debounce-input';
import PropTypes from 'prop-types';

const SearchBar = ({ query, handleChange }) => (
  <div className="search-books">
    <div className="search-books-bar">
      <Link className="close-search" to="/my-reads">
        Close
      </Link>
      <div className="search-books-input-wrapper">
        <DebounceInput
          minLength={1}
          debounceTimeout={300}
          type="text"
          placeholder="Search by title or author"
          value={query}
          onChange={handleChange}
        />
      </div>
    </div>
    <div className="search-books-results">
      <ol className="books-grid" />
    </div>
  </div>
);

SearchBar.propTypes = {
  query: PropTypes.string,
  handleChange: PropTypes.func.isRequired
};

export default SearchBar;
