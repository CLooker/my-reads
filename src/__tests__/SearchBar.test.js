import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from '../components/SearchBar';

describe('SearchBar', () => {
  const componentProps = {
    handleChange: jest.fn()
  };
  describe('code paths', () => {
    describe('`query` prop', () => {
      describe('`query` is provided', () => {
        it('should render correctly', () => {
          const localComponentProps = {
            ...componentProps,
            query: 'fake query'
          };

          const component = shallow(<SearchBar {...localComponentProps} />);

          expect(component).toMatchSnapshot();
        });
      });

      describe('`query` is not provided', () => {
        it('should render correctly', () => {
          const component = shallow(<SearchBar {...componentProps} />);

          expect(component).toMatchSnapshot();
        });
      });
    });
  });
});
