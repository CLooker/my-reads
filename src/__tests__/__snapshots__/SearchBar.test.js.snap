// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar code paths \`query\` prop \`query\` is not provided should render correctly 1`] = `
<div
  className="search-books"
>
  <div
    className="search-books-bar"
  >
    <Link
      className="close-search"
      replace={false}
      to="/my-reads"
    >
      Close
    </Link>
    <div
      className="search-books-input-wrapper"
    >
      <DebounceInput
        debounceTimeout={300}
        element="input"
        forceNotifyByEnter={true}
        forceNotifyOnBlur={true}
        minLength={1}
        onChange={[MockFunction]}
        placeholder="Search by title or author"
        type="text"
      />
    </div>
  </div>
  <div
    className="search-books-results"
  >
    <ol
      className="books-grid"
    />
  </div>
</div>
`;

exports[`SearchBar code paths \`query\` prop \`query\` is provided should render correctly 1`] = `
<div
  className="search-books"
>
  <div
    className="search-books-bar"
  >
    <Link
      className="close-search"
      replace={false}
      to="/my-reads"
    >
      Close
    </Link>
    <div
      className="search-books-input-wrapper"
    >
      <DebounceInput
        debounceTimeout={300}
        element="input"
        forceNotifyByEnter={true}
        forceNotifyOnBlur={true}
        minLength={1}
        onChange={[MockFunction]}
        placeholder="Search by title or author"
        type="text"
        value="fake query"
      />
    </div>
  </div>
  <div
    className="search-books-results"
  >
    <ol
      className="books-grid"
    />
  </div>
</div>
`;
